canvasobject.go

```diff
@@ -105,3 +109,85 @@ type Tabbable interface {
 type Tappable interface {
 	Tapped(*PointEvent)
 }
+
+type DragItem interface {
+	MimeType() string
+	URI() URI
+}
+
+type DragItemCursor interface {
+	// Image returns the image for the given cursor, or nil if none should be shown.
+	// It also returns the x and y pixels that should act as the hot-spot (measured from top left corner).
+	Image() (image.Image, int, int)
+}
+
+// DragSource is implemented by CanvasObjects to indicate that they can be dragged and dropped
+// onto another CanvasObject which implements DropTarget.
+type DragSource interface {
+	// DragSourceBegin is invoked when a drag and drop is initiated on this DragSource.
+	// It should return the DragItems it is sourcing, and optionally a non-nil DragItemCursor
+	// which will be used to visually represent the sourced content during the in-progress drag.
+	//
+	// Once a drag begins, either DropAccepted or DropCanceled
+	// will be invoked to notify this DragSource of the outcome of the drag.
+	// This is guaranteed to happen before another invocation of DragSourceBegin may be called.
+	DragSourceBegin(*PointEvent) ([]DragItem, DragItemCursor)
+
+	// DragSourceDragged is invoked when a drag and drop continues within this DragSource
+	// and has not yet entered over a potential DropTarget.
+	DragSourceDragged(*PointEvent)
+
+	// DropPending is invoked the first time an in progress drag and drop enters
+	// over a specific potential DropTarget that may accept the drop.
+	// The items that the drop target can accept are passed to this invocation,
+	// which MAY be a subset of the items returned by DragSourceBegin.
+	// Implementations may wish to use this hook to update rendering to
+	// highlight the item(s) which may be about to be dropped.
+	DropPending([]DragItem)
```

Comment:
Do you have a concrete example of this in use? I'm not entirely sure that it will be supported at the OS level...